src/components/Root/index.tsx

```diff
@@ -47,6 +47,9 @@ const RootConsumer = (): React.ReactElement | null => {
       }
     }
     initChains()
+
+    // Set store chainId and init contracts/session
+    setChainId(_getChainId())
```

Comment:
Does it make sense to call it when chains are loaded a few lines above? Or here is fine as well?

Reply:
It is already initiated but in order to trigger the middleware it needs to be called. It can either be here or in a seperate effect. It doesn't matter where.